.github/workflows/ci.yml

```diff
@@ -1,41 +1,54 @@
 name: CI
 
-on: [push]
+on:
+  push:
+  pull_request:
 
 jobs:
-  build-ubuntu:
+  build-and-test:
 
-    runs-on: ubuntu-latest
-    
-    steps:
-    - uses: actions/checkout@v2
-      with:
-          submodules: true
-    - name: configure
-      run: mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=debug ..
-    - name: build
-      run: cmake --build build
-    - name: test
-      run: cd build && ctest
+    name: ${{ matrix.toolchain }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        toolchain:
+          - linux-gcc
+          - macos-clang
+          - windows-msvc
 
+        configuration:
+          - Debug
+          # TODO: To be enabled in the future
+          # - Release
```

Comment:
Any reason you've left this one as `TODO`? Does it not work? I'd rather merge this either uncommented or removed entirely.

Reply:
This does not work because 
```
ignoring return value of ‘int system(const char*)’, declared with attribute warn_unused_result [-Werror=unused-result]
```

I'll remove this